lib/config/test_configuration.js

```diff
@@ -0,0 +1,135 @@
+var f = require('util').format;
+
+var Configuration = function(options) {
+  options = options || {};
+  var host = options.host || 'localhost';
+  var port = options.port || 27017;
+  var db = options.db || 'integration_tests';
+  var mongo = null;
+  var manager = options.manager;
+  var skipStart = typeof options.skipStart === 'boolean' ? options.skipStart : false;
+  var skipTermination = typeof options.skipTermination === 'boolean' ? options.skipTermination : false;
+  var setName = options.setName || 'rs';
+
+  // Default function
+  var defaultFunction = function(self, _mongo) {
+    return new _mongo.Server({
+      host: self.host,
+      port: self.port
+    });
+  };
+
+  // Create a topology function
+  var topology = options.topology || defaultFunction;
+
+  return function(context) {
+    mongo = require('mongodb-core');
+
+    return {
+      start: function(callback) {
+        var self = this;
+        if (skipStart) return callback();
+
+        // Purge the database
+        manager.purge().then(function() {
```

Comment:
I actually just fixed this in my local copy of `mongodb-core`s test runner. Promises should be returned back into the promise chain:

```
        // Purge the database
        var Logger = require('mongodb-topology-manager').Logger;
        manager.purge()
          .then(function() {
            console.log("[purge the directories]");
            return manager.start();
          })
          .then(function() {
            console.log("[started the topology]");

            // Create an instance
            var server = topology(self, mongo);
            console.log("[get connection to topology]");

            // Set up connect
            server.once('connect', function() {
              console.log("[connected to topology]");
              // Drop the database
              server.command(f("%s.$cmd", self.db), {dropDatabase: 1}, function(err, r) {
                console.log("[dropped database]");
                server.destroy();
                callback();
              });
            });

            // Connect
            console.log("[connecting to topology]");
            server.connect();
          })
          .catch(function(err) {
            console.log(err.stack);
          });
```